app/models/user.rb

```diff
@@ -306,6 +309,17 @@ def anonymous?
     self == User.anonymous
   end
 
+  def set_project_ids(project_ids)
+    all_projects = Admin::Project.all
+    all_projects.each do |project|
+      if project_ids.find { |id| id.to_i == project.id }
+        projects << project if !projects.include? project
+      else
+        projects.delete project
+      end
+    end
+  end
+
```

Comment:
I guess an easy way would be to use build-in magic `<association_name>_ids` setter: 
`self.project_ids = new_ids` 
(probably directly in the controller)
I can see that it's based on roles approach, not worth changing, just fyi.